hs628_tools/Test.py

```diff
@@ -0,0 +1,86 @@
+
+import numpy as np
+from sklearn import datasets
+import csv
+import os
+
+
+
+
+class data_split:
+    """
+    Create class data_split to predict a perfect train and test data split ratio
+
+    define methods to accomplish this task
+
+    return a split ratio
+
+    """
+    def __init__(self, data):
+        self.data = data
+
+    #set path to read the csv file in the directory
+    def test_train():
+
+        pathName = os.getcwd()
+
+        dataset = []
+        fileNames = os.listdir(pathName)
+
+        # read file with extension .csv
+        for fileNames in fileNames:
+            if fileNames.endswith(".csv"):
+                dataset.append(fileNames)
+
+        for k in dataset:
+            file = open(os.path.join(pathName, k), "rU")
+            data = csv.reader(file, delimiter=',')
+            # data_len =sum(1 for line in data)
+            # print data_len
+
+        clean_data = []
+        for column in data:
+
+            for row in column:
+
+                if row == "?":
+                    pass
+                if row == "NA":
+                    pass
+                if row == "NAN":
+                    pass
+
+                else:
+                    clean_data.append(row)
+
+
+        return clean_data
+
+        #********
+
+    test_train()
+
+    row_len = sum(1 for line in test_train())
+    print row_len
+
+
+    def split_ratio(row_len):
+        """
+
+        :return: data split ratio based on the number of data points
+        """
```

Comment:
It would be helpful to add in the docstrings what data type the output is that you are returning. I believe your code right now is just print statements so not returning any data type. I think it would be useful to instead return the ratio into a list especially for my visualization module because I want to access the test ratio calculated from this module in order to split the dataset and run my code.

Reply:
Hi Nikita,
Thank you for your suggestion and insight. I have updated my new codes with docstrings. I will make sure to word the docstring in accordance with what the codes actually do. That makes sense. I have completely changed my codes to have it print new train and test dataframes for the users. Please take a look and let me know if I can do anything to polish it. Thanks again!

Nirvesh